src/frontends/pytorch/src/op/cat.cpp

```diff
@@ -56,6 +58,38 @@ OutputVector translate_cat_common(const NodeContext& context,
             context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(shape_sliced, axis_const, neg_1, zero));
         return {context.mark_node(std::make_shared<v1::Reshape>(tensor, new_shape, false))};
     }
+
+    const auto first_in_type = list_elems.front().get_element_type();
+    const bool is_mixed_type =
+        list_elems.size() > 1 && (std::any_of(std::next(list_elems.begin()),
+                                              list_elems.end(),
+                                              [&first_in_type](const ov::Output<ov::Node>& input) {
+                                                  return input.get_element_type() != first_in_type ||
+                                                         input.get_element_type() == ov::element::dynamic;
+                                              }));
+    if (is_mixed_type) {
+        auto node_of_type = list_elems[0];
+        for (size_t i = 1; i < list_elems.size(); ++i) {
+            node_of_type =
+                std::make_shared<ov::op::v14::ConvertPromoteTypes>(node_of_type, list_elems[i], true)->output(0);
+            context.mark_node(node_of_type.get_node_shared_ptr());
+        }
+
+        const auto unified_type = node_of_type.get_element_type();
+        auto inputs_vec = OutputVector(list_elems.begin(), list_elems.end());
+        for (size_t i = 0; i < inputs_vec.size(); ++i) {
+            context.mark_node(inputs_vec[i].get_node_shared_ptr());
```

Comment:
```suggestion
```
`mark_node` should only be called for new nodes, here you create no nodes

Reply:
Okay, removed.